### ISSUE TITLE
False positive for `Style/SafeNavigation` when safe guarding arithmetic operation or assignment

### ISSUE THREAD

**Issue by tiagotex** [bbatsov/rubocop#5089]
`Style/SafeNavigation` returns false positives and tries to autocorrect when safeguarding a arithmetic operation or assignment on a chained method call.

--------

## Expected behavior

Rubocop shouldn't consider this an offense, since we don't want to do the operation when object is null.

## Actual behavior

Rubocop removes the safe guard and uses the safe operator.

## Steps to reproduce the problem

```ruby
foo.baz + bar if foo
foo.baz = bar if foo
```

 Rubocop tries to fix to:

```ruby
foo&.baz + bar
foo&.baz = bar
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin16)
```

**Comment by Drenmi** [comment]
Same issue as #5079. The problem seems to be that the cop is not checking that the **last** method in a method chain is a "dot-less" method (comparisons, arithmetics, hash access, etc.)

**Comment by tiagotex** [comment]
I'm working on a PR to fix it.

**Comment by tiagotex** [comment]
I fixed the problems described in this issue, but not the problem referenced in #5079, I might take a stab at it later.
